featuretools/primitives/standard/transform/natural_language/num_words.py

```diff
@@ -19,12 +28,21 @@ class NumWords(TransformPrimitive):
 
     name = "num_words"
     input_types = [ColumnSchema(logical_type=NaturalLanguage)]
-    return_type = ColumnSchema(semantic_tags={"numeric"})
+    return_type = ColumnSchema(logical_type=IntegerNullable, semantic_tags={"numeric"})
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
     description_template = "the number of words in {}"
 
     def get_function(self):
         def word_counter(array):
-            return array.fillna("").str.count(" ") + 1
+            def _get_number_of_words(elem: pd.Series):
+                """Returns the number of words in given element,
+                or null given non-iterable input"""
+                if not isinstance(elem, Iterable):
```

Comment:
Can't we just check if `elem` is of type `str` here instead without importing Iterable? Since the input of this primitive is a Natural Language column with a `string` dtype, I think the only possible options here are `str` or `pd.NA`. If you change this, please also update the docstring accordingly.